src/api/index.js

```diff
@@ -0,0 +1,32 @@
+const fakeDatabase = {
+  users: {
+    'admin@admin': { 'name': 'Admin', 'password': 'admin' }
+  }
+}
+
+const fakeRequest = () => new Promise(resolve => setTimeout(resolve, 500))
+
+export const authenticate = (email = '', password = '', keepAlive = false) =>
+  fakeRequest().then(() => {
+    let user
+    if (email && password) {
+      const record = fakeDatabase.users[email]
+      if (record && record.password === password) {
+        user = { name: record.name }
+        window.localStorage.setItem('user', JSON.stringify(user))
+      }
+    } else {
+      const userLocalStorage = window.localStorage.getItem('user')
+      if (userLocalStorage) {
+        user = JSON.parse(userLocalStorage)
+      }
+    }
+
+    return (user) ? { status: 200, user } : { status: 401 }
```

Comment:
The response object should implement the interface of the response object we will use in the future. If we decide to follow the Fetch API, then it should have a `json()` method which returns a promise with the body as json. As for the `status` property it is according to the fetch api response interface.
See: https://developer.mozilla.org/en-US/docs/Web/API/Response